xclim/indices/generic.py

```diff
@@ -925,17 +925,32 @@ def get_zones(
     step_size: Quantified
         Size of zones
     zone_1 : Quantified
-        Left boundary of the zone labelled as "zone 1"
+        Left boundary of the zone labelled as "zone 1". It should be in the range [`zone_min`, `zone_max` [.
     allow_boundary_zones : Bool
         Determines whether a zone value is attributed for values in ]-np.inf, zone_min[ and [zone_max, np.inf[.
     close_last_zone_right_boundary : Bool
         Determines if the right boundary of the last zone is closed.
     """
-    zone_1 = zone_1 or zone_min
-    zone_next = f"{str2pint(zone_min).magnitude + str2pint(step_size).magnitude}  {pint2cfunits(str2pint(zone_min))}"
+    zone1 = zone_1 or zone_min
+
+    mn_p, mx_p, step_p, z1_p = (
+        str2pint(v) for v in [zone_min, zone_max, step_size, zone1]
+    )
+    units = [pint2cfunits(p) for p in [mn_p, mx_p, step_p, z1_p]]
+    if len(set(units)) != 1:
+        msg = f"Got different units among `zone_min` ({units[0]}), `zone_max` ({units[1]}), `step_size` ({units[2]})"
+        warnings.warn(msg + (f", `zone_1 ({units[3]})`." if zone_1 else "."))
```

Comment:
Great!

Reply:
Why a warning ? Isn't this fully supported by line 94 below ?

Reply:
I asked Éric to add this. While it's fully supported (and who knows, maybe applicable to some studies?), I thought it would be good to have a warning for users to be aware that they're mixing units in their thresholds.

Reply:
Now that you ask, I'm also on the fence regarding the need for a warning. There are some instances where the same kind of behaviour is allowed in xclim. In this specific case, I do agree though, having different units for such closely related variables would make you wonder if the user has made an error of some sort. Perhaps it's not the kind of warning we want to see in xclim though.

As a comparison, `heat_wave_XYZ` has two thresholds (`thresh_tasmax` & `thresh_tasmin`). Although giving two thresh in "degC" and "degF" would be weird, it does not raise any warning.

Reply:
I think a user error would look like a misspelled unit (genre :  `deg C`), but that would raise an error anyway. Passing different units seems to me much more likely to be deliberate.

Reply:
Fair enough. We don't do that elsewhere, really. I think Pascal is right that we need to trust that the user knows what they're doing.

Sorry Éric, thanks again for adding it, but we should probably roll that back.